site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,1032 @@
+The value of this property is set to the project's `artifactId` as defined in its POM file and will be used later in the test code to allow Arquillian to access the application. +
+For that reason it is crucial that the value of the `app.name` property matches exactly the name of the `Route` used for the application under test.
+
+At this point it's imperative to note that when use the FMP defaults, all the generated Openshift resources use `project.artifactId` as their name. That is why
+we specified `project.artifactId` as the value for `app.name`. +
+Had we configured FMP to generate a Route using different name, say `myapp-route`, then we would also have to configure `app.name` with the value of  `app.route`
+
+[NOTE]
+====
+Although providing details for changing FMP defaults is out of the scope of this guide, we will nonetheless mention a simple way of configuring the generated Route's name.
+
+We can add link:https://maven.fabric8.io/#resource-fragments[resource fragments] that FMP will use to override the defaults it uses. A resource fragment for `Route` could be:
+
+.src/main/fabric8/route.yml
+----
+apiVersion: v1
+kind: Route
+metadata:
+  name: myapp-route
+spec:
+  port:
+    targetPort: 8080
+  to:
+    kind: Service
+    name: ${project.artifactId}
+----
+====
+
+
+<4> We configure two goals for the failsafe plugin, the `integration-test` and `verify` goals. By configuring these goals, the Maven surefile plugin will automatically
+participate in the corresponding Maven default lifecycle phases. See link:https://maven.apache.org/surefire/maven-failsafe-plugin/plugin-info.html[this] part of the failsafe documentation for more details.
+
+The `integration-test` goal is responsible for actually running the integration tests while `verify` is used in order to make sure that all tests passed (and fail the build if they don't).
+
+[NOTE]
+====
+It's also worth noting that we haven't specified which tests are considered integration tests. Omitting such link:https://maven.apache.org/surefire/maven-failsafe-plugin/examples/inclusion-exclusion.html[configuration]
+means that we are relying on failsafe's default behavior, which is to include all tests that satisfy any of the following patterns:
+
+* `"**/IT*.java"`
+* `"**/*IT.java"`
+* `"**/*ITCase.java"`
+====
+
+==== Arquillian Configuration
+
+Next comes adding the Arquillian configuration file which will configure Arquillian to interact with the OpenShift environment in the desired way.
+
+.src/test/resources/arquillian.xml
+----
+<arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns="http://jboss.org/schema/arquillian"
+            xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
+
+  <extension qualifier="openshift"> #<1>
+    <property name="namespace.use.current">true</property> #<2>
```

Comment:
Is the purpose of this guide to explain HTTP booster's IT setup or to suggest a good practice? If the latter, then I think by default we should show `namespace.use.current=false` (which is also a default) because creating a new environment for each run guarantees that environment is the same every time.

Reply:
Also, it might be good to mention that all these properties can be skipped and AFAIK the whole `arquillian.xml` file is optional.

Reply:
The purpose of the guide is to explain the various options  that are possible to achieve integration testing and provide some suggestions. 
To that end I have a warning note in another part of the guide.

I also have also noted earlier that arquillian uses plenty of defaults but I will also make it more explicit here